GoogleForCreators/web-stories-wp
# UI Simplification: Library insertion icon/menu

## Context



## Summary
- Adds an insertion overlay to each library element which has a scrim and a non-functional plus icon, or plus icon that opens an action menu
- Drag-drop still works
- Clicking anywhere other than the plus icon still inserts the element


## Relevant Technical Choices



## To-do

- [x] Refactor the new nested roving tabIndex util to combine it with the default one
- [x] Review all style
- [x] Rework Saved Page Templates to have two buttons
- [x] Adjust tests
- [x] Add new tests for the new menu items



## User-facing changes
<img width="339" alt="Screenshot 2022-02-01 at 19 12 18" src="https://user-images.githubusercontent.com/3294597/152016713-ce7bc280-4d8e-4b96-88c3-1db6ba1fbf9c.png">
<img width="339" alt="Screenshot 2022-02-01 at 19 12 11" src="https://user-images.githubusercontent.com/3294597/152016718-e6480f9e-e90f-4c2a-b956-0144d8a65642.png">
<img width="170" alt="Screenshot 2022-02-18 at 12 14 20" src="https://user-images.githubusercontent.com/3294597/154665062-2da9ed48-3df7-414c-83a9-58cc61d8254c.png">
<img width="331" alt="Screenshot 2022-02-18 at 12 14 05" src="https://user-images.githubusercontent.com/3294597/154665069-aa8842f5-2aa1-4d46-a8e2-084f15ed33e5.png">
<img width="255" alt="Screenshot 2022-02-18 at 12 13 58" src="https://user-images.githubusercontent.com/3294597/154665077-702e11d9-ccea-4c8f-be53-418ab4ff1ced.png">



## Testing Instructions






This PR can be tested by following these steps:

1. Open the Local Media tab
2. Verify that hovering over media will reveal a two menu buttons (previously it was just one -- the More button in the right top corner). Verify that clicking on the Insertion menu button will open a menu with two options: Inserting / Using as background. Verify these work as expected.
3. Open 3rd party Media tab.
4. Verify that hovering over media will reveal a menu button -- clicking on it opens dropdown with two options: Inserting / Using as background. Verify these work as expected.
5. Verify that for Stickers, Shapes, Default Page Templates, Text Presets and Text Sets, the plus icon appears when hovering but it will insert the element right away instead of opening a menu.
6. For Saved Page Templates -- verify the plus icon uses the template and Delete icon will open the Delete confirmation dialogue.
7. For all element types: Verify that dragging the media from around the menu button is still possible, as is clicking outside of the menu button for inserting right away.
8. For all elements: Verify that keyboard accessibility works as expected (please see details per element in the [original issue description](https://github.com/GoogleForCreators/web-stories-wp/issues/10109) -- it's written out per each element under the "Accessibility" point)


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10109 
Fixes #9884
